hibernate-cache/src/main/java/org/hibernate/cache/infinispan/access/LockingInterceptor.java

```diff
@@ -32,43 +34,41 @@
  */
 public class LockingInterceptor extends NonTransactionalLockingInterceptor {
 	private static final Log log = LogFactory.getLog(LockingInterceptor.class);
-	private static final boolean trace = log.isTraceEnabled();
 
-	@Override
-	protected Object visitDataWriteCommand(InvocationContext ctx, DataWriteCommand command) throws Throwable {
-		Object returnValue = null;
-		try {
-			// Clear any metadata; we'll set them as appropriate in TombstoneCallInterceptor
-			command.setMetadata(null);
+   protected final InvocationFinallyAction unlockAllReturnCheckCompletableFutureHandler = new InvocationFinallyAction() {
+      @Override
+      public void accept(InvocationContext rCtx, VisitableCommand rCommand, Object rv, Throwable throwable) throws Throwable {
+         lockManager.unlockAll(rCtx);
+         if (rv instanceof CompletableFuture) {
+            try {
+               ((CompletableFuture) rv).join();
```

Comment:
Hmmm, not sure I understand what the code should look like. Could you be more precise?